chore: add list of aurora engine versions tested (#382)
awslabs/aws-advanced-jdbc-wrapper: README.md

We enforce a style using [Google checkstyle](https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml).
This can be verified in the tests; tests will not pass with checkstyle errors.

## Aurora Engine Version Testing

> This driver has been tested against MySQL 5 and MySQL 8s.

Reviewer: This might be unintentionally misleading. Can we explicitly state the MySQL versions we tested against and the Aurora engine versions in our integration pipelines?

> This driver has been tested against MySQL 5 and MySQL 8s.

Reviewer 2: ```suggestion
| Database          | Versions                                                                                                                                                                                                         |
|-------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| MySQL             | 8.0.32                                                                                                                                                                                                           |
| PostgreSQL        | 15.2                                                                                                                                                                                                             |
| Aurora MySQL      | 8.0.mysql_aurora.3.02.2 (Wire-compatible with MySQL 8.0.23 onwards, more details see https://docs.aws.amazon.com/AmazonRDS/latest/AuroraMySQLReleaseNotes/AuroraMySQL.Updates.3022.html)                         |
| Aurora PostgreSQL | 14.5 (Compatible with PostgreSQL 14.5, see release notes [here](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraPostgreSQLReleaseNotes/AuroraPostgreSQL.Updates.html#AuroraPostgreSQL.Updates.20180305.145X)) |
```

## Releases

> The `aws-advanced-wrapper-driver` has a regular monthly release cadence. A new release will occur during the last week of each month. However, if there are no changes since the latest release, then a release will not occur.

Reviewer: Outside of this review, but could you fix this?
```suggestion
The `aws-advanced-jdbc-wrapper` has a regular monthly release cadence. A new release will occur during the last week of each month. However, if there are no changes since the latest release, then a release will not occur.
```

## Aurora Engine Version Testing

> This driver is being tested against the following versions in the current test suite

Reviewer: ```suggestion
This `aws-advanced-jdbc-wrapper` is being tested against the following Community and Aurora database versions in our test suite:
```

|-------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| MySQL             | 8.0.32                                                                                                                                                                                                           |
| PostgreSQL        | 15.2                                                                                                                                                                                                             |
| Aurora MySQL      | MySQL	8.0.mysql_aurora.3.02.2 (Wire-compatible with MySQL 8.0.23 onwards. For more details see [here](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraMySQLReleaseNotes/AuroraMySQL.Updates.3022.html))                     |
| Aurora PostgreSQL | 14.5 (Compatible with PostgreSQL 14.5, see release notes [here](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraPostgreSQLReleaseNotes/AuroraPostgreSQL.Updates.html#AuroraPostgreSQL.Updates.20180305.145X)) |

> The driver is compatible with MySQL 5 and MySQL 8 as per the Community MySQL JDBC Driver.

Reviewer: While MySQL support is experimental at the moment, the `aws-advanced-jdbc-wrapper` is compatible with MySQL 5.7 and MySQL 8.0 as per the Community MySQL Connector/J 8.0 Driver.
